source/server/listener_impl.cc

```diff
@@ -512,11 +531,15 @@ void ListenerImpl::buildAccessLog() {
 }
 
 void ListenerImpl::buildInternalListener() {
-  if (config_.address().has_envoy_internal_address()) {
+  if (config_.has_internal_listener()) {
+    if (config_.has_address()) {
```

Comment:
this check https://github.com/envoyproxy/envoy/blob/c4837b52cebc46022e536061755ee29797c97c32/source/server/listener_manager_impl.cc#L386-L394

is useless anymore.

or we can move this check to 
https://github.com/envoyproxy/envoy/blob/c4837b52cebc46022e536061755ee29797c97c32/source/server/listener_manager_impl.cc#L387
and replace the original check, then it will avoid ListenerImpl's construct to do the useless parsing for the internal addresses
https://github.com/envoyproxy/envoy/pull/22274/files#diff-b0d5e1a1317df8e22a556ee74d6f728379c5f85ce640a9404acc14bfe8192ad3R371